third_party/move/move-model/bytecode/src/stackless_bytecode.rs

```diff
@@ -309,6 +309,72 @@ impl Operation {
             Operation::TraceGlobalMem(..) => false,
         }
     }
+
+    /// Returns true if the operation can take primitive non reference values as arguments.
+    pub fn can_take_non_ref_primitive_as_args(&self) -> bool {
+        match self {
+            Operation::Function(_, _, _) => true,
+            Operation::OpaqueCallBegin(_, _, _) => false,
+            Operation::OpaqueCallEnd(_, _, _) => false,
+            Operation::Pack(_, _, _) => false,
+            Operation::Unpack(_, _, _) => false,
+            Operation::MoveTo(_, _, _) => false,
+            Operation::MoveFrom(_, _, _) => false,
+            Operation::Exists(_, _, _) => false,
+            Operation::BorrowLoc => false,
+            Operation::BorrowField(_, _, _, _) => false,
+            Operation::BorrowGlobal(_, _, _) => false,
+            Operation::GetField(_, _, _, _) => false,
+            Operation::GetGlobal(_, _, _) => false,
+            Operation::Uninit => false,
+            Operation::Drop => false,
```

Comment:
Can't this take primitive values?

Reply:
That seems like a current implementation choice - but there is nothing limiting `Drop` from taking primitive values as arguments?

Reply:
Yeah, you are right but I am not sure how to write a source code level test for the `Drop` case, any suggestions on how to test it?

Reply:
Unfortunately, I don't think we can write a source code level test for that case.

Reply:
Added a comment to explain why we currently return false for `Drop`